packages/core/admin/ee/server/services/audit-logs/process-event.js

```diff
@@ -0,0 +1,33 @@
+'use strict';
+
+module.exports = function processEvent(strapi, name, ...args) {
```

Comment:
It would be nice to know if we need to spread `...args` or not. I get the feeling this parameter could just be called `payload`. Then we wouldn't need the `defaultGetPayload` function

Reply:
Currently it's not used, but we allow passing as many args as wanted in the event hub, so if we don't spread here we lose access to everything but the 1st arg. I think if we wanted to prevent passing several args we should have done it in the event emitter, but we wanted to follow the node event emitter behavior there

Reply:
The 1 place where we could easily remove args[0] is the default get payload:

```js
const getDefaultPayload = payload => payload
```

looks a bit weird but it will work

Reply:
If we want to keep as simple as it can be that also works for me. But in that case I think we can remove the object altogether and just keep an array of allowed events, since there's no point in having 1 function per event anymore

Reply:
> I think if we wanted to prevent passing several args we should have done it in the event emitter, but we wanted to follow the node event emitter behavior there 

The eventhub is used elsewhere in the application and is not specific to audit logs, so I think it still makes sense to accept many arguments there. It seems to me (at least so far) we will always just pass a payload for audit logs.

I guess though I'm having a bit of a tough time understanding when we would need additional arguments here for audit logs.

So far though I think the map does look like a bit of overkill because they all accept the same `getDefaultPayload` and that function itself doesn't do much. @remidej maybe you could provide an example for what you see in the future?

I do think providers should be able to extend this map with their own function, but not sure that is dependant on the `...args` argument. 

I brought this up because still wasn't clear to me why we needed it and I'll be off for a while. But either way we can always change it later even after audit logs is released, I don't think it would be a breaking changing. So we can keep moving forward either way in my opinion.